src/compton/Compton_NWA.cc

```diff
@@ -0,0 +1,138 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   compton/Compton_NWA.cc
+ * \author Kendra Keady
+ * \date   Tues Feb 21 2017
+ * \brief  Implementation file for compton NWA interface
+ * \note   Copyright (C) 2017 Los Alamos National Security, LLC.
+ *         All rights reserved.
+ */
+
+// headers provided in draco:
+#include "Compton_NWA.hh"
+#include "ds++/Assert.hh"
+// headers provided in Compton_NWA include directory:
+#include "compton_file.hh"
+#include "multigroup_compton_data.hh"
+#include "multigroup_data_types.hh"
+#include "multigroup_lib_builder.hh"
+
+namespace rtt_compton {
+
+// ------------ //
+// Constructors //
+// ------------ //
+
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for an existing multigroup libfile.
+ *
+ * This calls NWA methods to read the data file and store everything in a
+ * Compton data object, a smart pointer to which is then passed to (and held by)
+ * the NWA etemp_interp class.
+ *
+ * \param filehandle The name of the multigroup file to use for Compton scatters
+ */
+Compton_NWA::Compton_NWA(const std::string &filehandle) {
+
+  // Make a compton file object
+  compton_file Cfile(false);
+
+  // read the (existing multigroup) library into a compton data container
+  std::shared_ptr<multigroup_compton_data> Cdata =
+      Cfile.read_mg_csk_data(filehandle);
+
+  // be sure the Cdata pointer is non-null:
+  Ensure(Cdata);
+
+  // initialize the electron temperature interpolator with the mg compton data
+  ei.reset(new etemp_interp(Cdata));
+
+  // Make sure the SP exists...
+  Ensure(ei);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for an existing pointwise file and a multigroup structure.
+ *
+ * This calls NWA methods to read the pointwise library and construct a
+ * multigroup Compton data object, a smart pointer to which is then passed
+ * to (and held by) the NWA etemp_interp class.
+ *
+ * \param filehandle The name of the pointwise lib to build MG data from
+ * \param grp_bds    A vector containing the multigroup bounds (in keV)
+ * \param n_xi       The number of angular points/Legendre moments desired
+ */
+Compton_NWA::Compton_NWA(const std::string &filehandle,
+                         const std::vector<double> &grp_bds, const size_t nxi) {
+
```

Comment:
* Can anything in the constructor body be moved to a constructor initialization list?
* Add `Require` DbC to check inputs?